### ISSUE TITLE
Add support for JSONField

### ISSUE THREAD

**Issue by User3759685** [incuna/django-pgcrypto-fields#103]
We have some data in JSON-field (django.contrib.postgres.fields.JSONField), but this package does not support it. Could it be possible to add this functionality?

**Comment by peterfarrell** [comment]
@User3759685 Not sure this can actually be implemented. Due to the encryption, all Django-PGCrypto-Fields are stored in the DB as `text` fields (unlimited character length) including Date, DateTime, etc. The JSON field provided by Django is persisted at the DB as a `jsonb` field.

You could store encrypted JSON as a `text` field but you would lose all operators at the database level to search it (`__has_key`, etc.).   Currently, there is nothing stopping you from storing JSON in a `TextPGPSymmetricalKeyField` and decoding the JSON into a native Python data types.

I question the usefulness of adding direct JSON support in this library.  Please let us know if you want to work on a PR for this feature.

**Comment by december1981** [comment]
I agree it's probably not worth integrating directly, but I needed the basic convenience of json encoding/decoding on top of an encrypted text field (to make code run seamless whether I decide in the end to encrypt the field or not - adapting perhaps only field queries which may be "jsonb aware" or simply raw "text contains").

So here's a gist (JSONTextFieldMixin class based on what Django JSONField does, without the "KeyTransform" gubbins):
https://gist.github.com/december1981/ba2a3bb281f7fd4428a178f1d228e68b

**Comment by peterfarrell** [comment]
@some1ataplace PRs for a new JSON field are welcome.
